clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -528,20 +527,33 @@ object GarbageCollector {
   ) = {
     val reportLogsDst = concatToGCLogsPrefix(storageNSForHadoopFS, "summary")
     val reportExpiredDst = concatToGCLogsPrefix(storageNSForHadoopFS, "expired_addresses")
+    val deletedObjectsDst = concatToGCLogsPrefix(storageNSForHadoopFS, s"deleted_objects")
+
     val time = DateTimeFormatter.ISO_INSTANT.format(java.time.Clock.systemUTC.instant())
     writeParquetReport(commitsDF, reportLogsDst, time, "commits.parquet")
-    writeParquetReport(expiredAddresses, reportExpiredDst, time)
+    try {
+      writeParquetReport(expiredAddresses, reportExpiredDst, time)
+      val expiredDF = spark.read.parquet(f"${reportExpiredDst}/dt=${time}/")
```

Comment:
wouldn't it be sufficient to read the `expiredAddresses` DF?
I think that the source of truth for the marked addresses is: `/_lakefs/retention/gc/addresses/mark_id=<>/`

Reply:
I just didn't want to take any risks with trying to cache this possibly huge DataFrame (or worse, re-computing it). Therefore I chose to read the Parquet from the storage. Thoughts?

Reply:
i'm also not sure how is the current approach of re-reading expired addresses from storage is different than what we do in readExpiredAddresses. can you please help me understand?

Reply:
readExpiredAddresses will read the addresses from a previous, mark-only run. This is not necessarily the case here.

Reply:
Not quite sure what you mean by "caching or recomputing the DF". 
You anyway write the `expiredAddresses` DF in line 535 which means that it will be computed, no?

Reply:
@Jonathan-Rosenberg - right. We may have another performance bug here, I just didn't want to add another one.

Reply:
Only getting here today, sorry.

It's really annoying.  I think fastest would be to cache the DF before writing it, and then count() that.  But then you need local disk space on the workers.  Rereading Parquet might be slower because I think it will need to open each object written.  But not by a lot: IIRC Parquet has a row count in every file

SO has several answers, all very old, none particularly informative.  You can pipe the results through something that counts, but that is quite intrusive.